File: analyzer.py
import stringdata,time

def linear_search(container, element: str) -> int:
    for index, thing in enumerate(container):
        if element == thing:
            return index
    return -1

def binary_search(container, element: str):
    low = 0
    high = len(container) - 1
    mid = 0

    while low <= high:
        mid = (high + low) // 2
        new_string=container[mid]
        for i,letter in enumerate(new_string):
            if letter == element[i]:
                if i == 4:
                    return mid
                continue
            if letter<element[i]:
                low=mid+1
                break
            elif letter>element[i]:
                high=mid-1
                break

    return -1



def main():
    data=stringdata.get_data()

    times = time.time()
    print("Index of not_here",linear_search(data, "not_here"))
    print("Linear Search took:", time.time() - times, "Seconds for case not_here")
    times = time.time()
    print("Index of not_here",binary_search(data, "not_here"))
    print("Binary Search took:", time.time() - times, "Seconds for case not_here")

    print("")

    times=time.time()
    print("Index of mzzzz",linear_search(data,"mzzzz"))
    print("Linear Search took:",time.time()-times, "Seconds for case mzzzz")
    times=time.time()
    print("Index of mzzzz",binary_search(data,"mzzzz"))
    print("Binary Search took: %.1f" % (times-time.time()), "Seconds for case mzzzz")

    print("")

    times=time.time()
    print("Index of aaaaa", linear_search(data, "aaaaa"))
    print("Linear Search took:", time.time() - times, "Seconds for case aaaaa")
    times = time.time()
    print("Index of aaaaa",binary_search(data, "aaaaa"))
    print("Binary Search took:", time.time() - times, "Seconds for case aaaaa")




if __name__ == '__main__':
    main()

File: analyzer.cpp
#include <iostream>

#include "StringData.h"


int linear_search(std::vector<std::string> data, std::string input) {
    for(int i=0; i<data.size(); i++) {
        if(data.at(i) == input) {
            return i;
        }
    }
    return -1;
}

int binary_search(std::vector<std::string> container, std::string input) {
    int low = 0;
    int high = container.size() - 1;
    int mid = 0;

    while (low <= high) {
        mid = (high + low);
        std::string new_string=container[mid];
        for(int i=0; i<new_string.size(); i++) {
            if (new_string[i] == input[i]) {
                if(i==4) {
                    return mid;
                }
                continue;
            }
            if (new_string[i] < input[i]) {
                low=mid+1;
                break;
            }
            if (new_string[i] > input[i]) {
                high=mid-1;
                break;
            }
        }
    }
    return -1;

}



int main(){
    std::vector<std::string> my_data=getStringData();
    long long time=systemTimeNanoseconds();
    std::cout << "Index of Linear Search for:'not_here'=" <<linear_search(my_data, "not-here") << std::endl;
    std::cout <<"That case took " <<time-systemTimeNanoseconds() <<" seconds"<< std::endl;

    time=systemTimeNanoseconds();
    std::cout << "Index of Binary Search for:'not-here'=" <<binary_search(my_data, "not-here") << std::endl;
    std::cout <<"That case took " <<time-systemTimeNanoseconds() <<" seconds"<< std::endl;

    time=systemTimeNanoseconds();
    std::cout << "Index of Linear Search for:'mzzzz'=" <<linear_search(my_data, "mzzzz") << std::endl;
    std::cout <<"That case took " <<time-systemTimeNanoseconds() <<" seconds"<< std::endl;

    time=systemTimeNanoseconds();
    std::cout << "Index of Binary Search for:'mzzzz'=" <<binary_search(my_data, "mzzzz") << std::endl;
    std::cout <<"That case took " <<time-systemTimeNanoseconds() <<" seconds"<< std::endl;

    time=systemTimeNanoseconds();
    std::cout << "Index of Linear Search for:'aaaaa'=" <<linear_search(my_data, "aaaaa") << std::endl;
    std::cout <<"That case took " <<time-systemTimeNanoseconds() <<" seconds"<< std::endl;

    time=systemTimeNanoseconds();
    std::cout << "Index of Binary Search for:'aaaaa'=" <<binary_search(my_data, "aaaaa") << std::endl;
    std::cout <<"That case took " <<time-systemTimeNanoseconds() <<" seconds"<< std::endl;


  return 0;
  }